### PR TITLE
[functions-framework-cpp] update to latest release (v0.4.0)

### PR DESCRIPTION
**Pull request by coryan**
**Describe the pull request**

Updates the `functions-framework-cpp` port to the latest release (v0.4.0)

- What does your PR fix? Fixes #

N/A

- Which triplets are supported/not supported? Have you updated the CI baseline?

No change.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by coryan** [comment]
I think the macOS build is stuck again? In any case, it passed earlier, this is ready for review.

File: ports/functions-framework-cpp/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "functions-framework-cpp",
-  "version": "0.3.0",
+  "version": "0.4.0",
   "port-version": 2,
```
Comment (JonLiu1993) [inline ports/functions-framework-cpp/vcpkg.json:4]: You should delete the port version

Comment (coryan) [reply ports/functions-framework-cpp/vcpkg.json]: I was wondering about that. Done.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks!

**Pull request merged**